notebooks/image_models/solutions/diffusion_model.ipynb

```diff
@@ -0,0 +1,723 @@
+{
```

Comment:
<p>Explain in more detail what the underlying process that's happening during training is so that we can understand what the <code>time</code>,<code>noise_rate</code>,and <code>signal_rate</code> are.  From the current wording, it sounds like this is a continuous diffusion process rather than a discrete, one: maybe clarify that point too.  I am having a hard time making sense of the current explanation, which I find a bit too short:</p><p> </p><blockquote>"We generate the noisy image (for the forward diffusion processes) by weighting random noise by the noise rate, and the training image by the signal rate, then adding them together. "</blockquote><p> </p><p>e.g. what do you mean by "weighting random noise by the noise rate" or "the training image by the signal rate"? </p><p> </p><p>I agree with @takumi about adding explanation about what the <code>diffusion_schedule()</code> is (and also why it's a function).  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/347/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Reply:
<p>Added more thorough explanation and visualization. </p>